module-system/sov-cli/src/workflows/transactions.rs

```diff
@@ -1,107 +1,129 @@
-                wallet_state.unsent_transactions.push(tx);
-            }
+            TransactionWorkflow::Import(import_workflow) => import_workflow.run(wallet_state),
             TransactionWorkflow::List => {
                 println!("Current batch:");
                 println!(
                     "{}",
                     serde_json::to_string_pretty(&wallet_state.unsent_transactions)?
                 );
+                Ok(())
+            }
+            TransactionWorkflow::Clean => {
+                wallet_state.unsent_transactions.clear();
+                Ok(())
+            }
+            TransactionWorkflow::Remove { index } => {
+                wallet_state.unsent_transactions.remove(index);
+                Ok(())
             }
         }
-
-        Ok(())
     }
 }
+/// An argument passed as path to a file
+#[derive(clap::Parser)]
+pub struct FileArg {
+    /// The path to the file
+    #[arg(long, short)]
+    pub path: String,
+}
 
 #[derive(clap::Subcommand)]
 /// Import a pre-formatted transaction from a JSON file or as a JSON string
-pub enum ImportTransaction {
+pub enum ImportTransaction<Json: Subcommand, File: Subcommand> {
     /// Import a transaction from a JSON file at the provided path
-    #[command(arg_required_else_help(true))]
-    FromFile {
-        /// The expected format of the file contents is: {"module_name": {"call_name": {"field_name": "field_value"}}}
-        path: PathBuf,
-    },
+    #[clap(subcommand)]
+    FromFile(Json),
     /// Provide a JSON serialized transaction directly as input
-    #[command(arg_required_else_help(true))]
-    FromString {
+    #[clap(subcommand)]
+    FromString(
         /// The JSON serialized transaction as a string.
         /// The expected format is: {"module_name": {"call_name": {"field_name": "field_value"}}}
-        json: String,
-    },
+        File,
+    ),
 }
 
-/// A wrapper around a subcommand that also includes the optional global arguments
-#[derive(clap::Parser)]
-pub struct TransactionSubcommand<S: Subcommand> {
-    /// The optional arguments
-    #[clap(flatten)]
-    pub args: OptionalArgs,
-    /// The inner subcommand
-    #[clap(subcommand)]
-    pub inner: S,
-}
+impl<Json, File> ImportTransaction<Json, File>
+where
+    Json: Subcommand,
+    File: Subcommand,
+{
+    /// Parse from a file or a json string
+    pub fn run<RT: CliWallet, C: sov_modules_api::Context, U, E1, E2, E3>(
+        self,
+        wallet_state: &mut WalletState<RT::Decodable, C>,
+    ) -> Result<(), anyhow::Error>
+    where
+        Json: CliFrontEnd<RT>,
+        File: CliFrontEnd<RT>,
+        Json: TryInto<RT::CliStringRepr<U>, Error = E1>,
+        File: TryInto<RT::CliStringRepr<U>, Error = E2>,
+        RT::CliStringRepr<U>: TryInto<RT::Decodable, Error = E3>,
+        RT::Decodable: Serialize,
+        E1: Into<anyhow::Error> + Send + Sync,
+        E2: Into<anyhow::Error> + Send + Sync,
+        E3: Into<anyhow::Error> + Send + Sync,
```

Comment:
Would it be ok to hardcode errors to anyhow in the upstream code? This way, we would reduce the number of generics needed by `run.`

Reply:
I think there should also be a way to get rid of these error types without hardcoding when we switch the generics. If not, I'll try to hardcode at least some of these to reduce the number of generics here.